### PR TITLE
feat(KtField): scope the isLoading skeleton to the input only

### PR DESCRIPTION
**Pull request by santiagoballadares**
Add isLoadingValue prop to show loading state only on the input area of the field

### CODE REVIEW

File: packages/kotti-ui/source/kotti-field/KtField.vue [hunk 1]
```diff
@@ -32,7 +32,10 @@
 				class="kt-field__help-description"
 				v-text="field.helpDescription"
 			/>
-			<div class="kt-field__input-container-wrapper">
+			<div
+				v-show="!field.isLoadingValue"
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-field/KtField.vue:36]: why the v-show?

**Comment by carsoli** [comment]
I don't understand why we need isLoading vs isLoadingValue...

Has the design decided we have two cases?
Also, there is certain implementation on the KtForm that makes the `isLoading` prop one of the inheritable props, so that a user app can pass `isLoading` on a `KtForm` and it would trickle down to all fields in that form contexts and update their `isLoading` here
..The logic would have to be replicated if we want to keep both props

I do, however, find it a bad decision to bloat the API like this.. Design needs to decided what they really want here, they claimed the label is barely affected by the async API requests, why are we keeping both implementations then?

File: packages/kotti-ui/source/kotti-field/KtField.vue (hunk 1 above)
Comment (santiagoballadares) [reply packages/kotti-ui/source/kotti-field/KtField.vue:36]: KtFieldDates have hooks that break with `v-if`

**Comment by carsoli** [comment]
> > I don't understand why we need isLoading vs isLoadingValue...
> > Has the design decided we have two cases? Also, there is certain implementation on the KtForm that makes the `isLoading` prop one of the inheritable props, so that a user app can pass `isLoading` on a `KtForm` and it would trickle down to all fields in that form contexts and update their `isLoading` here ..The logic would have to be replicated if we want to keep both props
> > I do, however, find it a bad decision to bloat the API like this.. Design needs to decided what they really want here.
> 
> Yes, design wants 2 different cases for this; and yes, I missed the KtForm stuff. So, drafting this for now.

I see you touched on the `inheritableProps` type
and `useInheritableProperties` 

I don't know why and if I was on an outdated version of the MR when I left the comment that concerned inheriting from the `KtForm` context

there is no issue with this as far as I can see.


Can you relay the reason why the design saw a reason to distinguish both scenarios?

**Comment by santiagoballadares** [comment]
> > > I don't understand why we need isLoading vs isLoadingValue...
> > > Has the design decided we have two cases? Also, there is certain implementation on the KtForm that makes the `isLoading` prop one of the inheritable props, so that a user app can pass `isLoading` on a `KtForm` and it would trickle down to all fields in that form contexts and update their `isLoading` here ..The logic would have to be replicated if we want to keep both props
> > > I do, however, find it a bad decision to bloat the API like this.. Design needs to decided what they really want here.
> > 
> > 
> > Yes, design wants 2 different cases for this; and yes, I missed the KtForm stuff. So, drafting this for now.
> 
> I see you touched on the `inheritableProps` type and `useInheritableProperties`
> 
> I don't know why and if I was on an outdated version of the MR when I left the comment that concerned inheriting from the `KtForm` context
> 
> there is no issue with this as far as I can see.
> 
> Can you relay the reason why the design saw a reason to distinguish both scenarios?

During grooming it was agreed that the loading was going to be changed to only show the loading bar on the value and not the labels but afterwards, design explained that they had from the beginning 2 different loading cases in mind.
Already complained about inaccurate AC during grooming.

For context, this is design's comment on this:

> We have 2 different cases for the loading:
> 
> 1. You are loading the entire field - e.g. on page load of a form, on switching tabs, on opening a drawer, etc -> no changes expected for this, the whole form field is in loading state (label, helptext, input, etc.)
> 
> 2. You are updating a value and a communication to BE is required in the specific moment - e.g. changing value of the inline field, changing a mail input with BE validation, etc. -> then ONLY the input field is in loading state
> 
> The ticket is targeting ONLY case 2, no changes should apply to case 1

**Comment by carsoli** [comment]
> > > > I don't understand why we need isLoading vs isLoadingValue...
> > > > Has the design decided we have two cases? Also, there is certain implementation on the KtForm that makes the `isLoading` prop one of the inheritable props, so that a user app can pass `isLoading` on a `KtForm` and it would trickle down to all fields in that form contexts and update their `isLoading` here ..The logic would have to be replicated if we want to keep both props
> > > > I do, however, find it a bad decision to bloat the API like this.. Design needs to decided what they really want here.
> > > 
> > > 
> > > Yes, design wants 2 different cases for this; and yes, I missed the KtForm stuff. So, drafting this for now.
> > 
> > 
> > I see you touched on the `inheritableProps` type and `useInheritableProperties`
> > I don't know why and if I was on an outdated version of the MR when I left the comment that concerned inheriting from the `KtForm` context
> > there is no issue with this as far as I can see.
> > Can you relay the reason why the design saw a reason to distinguish both scenarios?
> 
> During grooming it was agreed that the loading was going to be changed to only show the loading bar on the value and not the labels but afterwards, design explained that they had from the beginning 2 different loading cases in mind. Already complained about inaccurate AC during grooming.
> 
> For context, this is design's comment on this:
> 
> > We have 2 different cases for the loading:
> > 
> > 1. You are loading the entire field - e.g. on page load of a form, on switching tabs, on opening a drawer, etc -> no changes expected for this, the whole form field is in loading state (label, helptext, input, etc.)
> > 2. You are updating a value and a communication to BE is required in the specific moment - e.g. changing value of the inline field, changing a mail input with BE validation, etc. -> then ONLY the input field is in loading state
> > 
> > The ticket is targeting ONLY case 2, no changes should apply to case 1

even if the whole form is loading in terms of fetching data-from the backend on initial page/form load, the label strings are static translations all across our app

I think design is over-complicating the situation here 🤷🏼

**Review by carsoli — APPROVED** [review]

**Comment by carsoli** [comment]
@santiagoballadares made the changes as discussed with design and agreed with you

**Pull request merged**